### DISCUSSION TITLE
entity.id's can only be switches?

### DISCUSSION THREAD

**Discussion by morris2009** [jmcollin78/solar_optimizer#40] [Q&A]
Hi, I am wondering if, or why, the entity.id can only be a switch?
I have several powerplugs from the Philips HUE ecosystem and  they are all addressed as lights.
So the service_call wouldn't be switch.turn_on, but something like light.turn_on.

How could I make this work? what workaround should i try to get a device like a hue powerplug to work with solar_optimizer?
Are there pressing reason's to only allow switches as device.id ?

**Comment by jmcollin78** [comment]
In HA you can show a switch as a light or a switch. What is the case to turn light on for optimizing the Solar Production ?

**Reply by morris2009** [reply]
Because switches (smart plugs) from vendors like Philips Hue are always light entitites. No way of changing that apparently.
So i don't want to switch a light perse (although, if it are infrared heating lights it might still make sense to switch them on solar power).
This is about smart plugs that switch on/of fan's, mobile airco's and other stuff. 

In HA, a switch can be shown as light if  you want, but a light entity can't be shown as a switch.

**Comment by jmcollin78** [comment]
ok. Do you try to put your configuration as:

```
    action_mode: "service_call"
    activation_service: "light/turn_on"
    deactivation_service: "light/turn_off"
``` 

I don't test this but this should work out of the box.

In fact, mainly all services can be call with this mecanism.

**Reply by morris2009** [reply]
I thought i tried that, but i might have used light.turn_on instead of light/turn_on
Will try again
